### ISSUE TITLE
Regarding text_masks

### ISSUE THREAD

**Issue by xiankgx** [lucidrains/imagen-pytorch#203]
```
text_embeds = torch.randn(4, 256, 768).cuda()
images = torch.randn(4, 3, 256, 256).cuda()

# feed images into imagen, training each unet in the cascade

for i in (1, 2):
    loss = imagen(images, text_embeds = text_embeds, unet_number = i)
    loss.backward()
```

With refence to your example in the README. If text_embeds is padded to some maximum length or the longest length in a batch, do I also need to supply the text_masks?

**Comment by xiankgx** [comment]
Or is just forcing the embeddings corresponding to masked input locations to 0 as done in https://github.com/lucidrains/imagen-pytorch/blob/main/imagen_pytorch/t5.py#L104 sufficient?

**Comment by lucidrains** [comment]
@xiankgx yup, just pad with zeros and the framework will automatically take care of masking!

**Comment by xiankgx** [comment]
Ok, thanks.
